5.0.0-alpha.3: valid-jsdoc unexpectedly expects returns for async functions, even with requireReturn: false

<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).

    This template is for bug reports. If you are here for another reason, please see below:

    1. To propose a new rule: https://eslint.org/docs/developer-guide/contributing/new-rules
    2. To request a rule change: https://eslint.org/docs/developer-guide/contributing/rule-changes
    3. To request a change that is not a bug fix, rule change, or new rule: https://eslint.org/docs/developer-guide/contributing/changes
    4. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

    Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**Tell us about your environment**

* **ESLint Version:** 5.0.0-alpha.3
* **Node Version:** 8.11.2
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

<!-- Paste your configuration below: -->
```js
module.exports = {
  "rules": {
  },
  "env": {
    "es6": true
  },
  "parserOptions": {
    "ecmaVersion": 9,
  },
  "rules": {
    "valid-jsdoc": ["error", {
      "prefer": {
	"return": "returns",
      },
      "requireReturn": false,
      "requireReturnDescription": false,
    }]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

<!-- Paste the source code below: -->
```js
/**                                                                                                                                                       
 * A comment.                                                                                                                                             
 */
async function foo() {
  if (!this.bar) {
    return;
  }

  await this.bar();
}
```

<!-- Paste the command you used to run ESLint: -->
```bash
./node_modules/.bin/eslint eslint5.0test.js
```

**What did you expect to happen?**

No failures

**What actually happened? Please include the actual, raw output from ESLint.**

```bash
eslint4.0test.js
  1:1  error  Missing JSDoc @returns for function  valid-jsdoc

✖ 1 problem (1 error, 0 warnings)
```
This is a possible regression from #10161

--- a/lib/rules/valid-jsdoc.js
+++ b/lib/rules/valid-jsdoc.js
@@ -408,7 +408,7 @@ module.exports = {
                 if (!isOverride && !hasReturns && !hasConstructor && !isInterface &&
                     node.parent.kind !== "get" && node.parent.kind !== "constructor" &&
                     node.parent.kind !== "set" && !isTypeClass(node)) {
-                    if (requireReturn || functionData.returnPresent) {
+                    if (requireReturn || (functionData.returnPresent && !node.async)) {
                         context.report({
                             node: jsdocNode,
                             message: "Missing JSDoc @{{returns}} for function.",

--- a/tests/lib/rules/valid-jsdoc.js
+++ b/tests/lib/rules/valid-jsdoc.js
@@ -296,6 +296,17 @@ ruleTester.run("valid-jsdoc", rule, {
                 ecmaVersion: 2017
             }
         },
+        {
+            code:
+              "/**\n" +
+              " * An async function. Options do not require return.\n" +
+              " */\n" +
+              "async function a() {}",
+            options: [{ requireReturn: false }],
+            parserOptions: {
+                ecmaVersion: 2017
+            }
+        },
 
         // type validations
         {
@@ -1673,6 +1684,24 @@ ruleTester.run("valid-jsdoc", rule, {
             options: [{ requireReturn: false }],
             errors: [{
                 message: "JSDoc syntax error.",
+                type: "Block"
+            }]
+        },
+
+        // async function
+        {
+            code:
+              "/**\n" +
+              " * An async function. Options requires return.\n" +
+              " */\n" +
+              "async function a() {}",
+            output: null,
+            options: [{ requireReturn: true }],
+            parserOptions: {
+                ecmaVersion: 2017
+            },
+            errors: [{
+                message: "Missing JSDoc @returns for function.",
                 type: "Block"
             }]
         },

